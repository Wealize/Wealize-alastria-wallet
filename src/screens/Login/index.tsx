import React, { useState } from 'react'
import { NavigationFunctionComponent } from 'react-native-navigation'
import FlashMessage, { showMessage } from 'react-native-flash-message'

import {
  initStackWithScreen,
  NavigationProps,
  pushScreen
} from '../../utils/navigation-utils'
import LocalStorageService, {
  STORAGE_KEYS
} from '../../services/LocalStorageService'
import { checkPin } from '../../utils/keychain'
import {
  ContainerBottom,
  Title,
  InputsContainer,
  InputStyled,
  MainContainer,
  RecoveryButton
} from './styles'
import { Colors } from '../../utils/themes'
import { LOGIN, REGISTER } from '../../constants/text'
import TouchableButton from '../../components/TouchableButton'
import { SCREEN } from '../../constants/screens'

const Login: NavigationFunctionComponent = ({
  componentId
}: NavigationProps) => {
  const [pin, setPin] = useState('')

  const login = async () => {
    const IS_DID_CREATED = await LocalStorageService.getBool(
      STORAGE_KEYS.IS_DID_CREATED
    )

    if (IS_DID_CREATED) {
      initStackWithScreen(SCREEN.ACCREDITATION_LIST)
    } else {
      pushScreen(componentId, SCREEN.ALASTRIA_TOKEN_QR_READER)
    }
  }

  const showError = (): void => {
    showMessage({
      message: LOGIN.ERROR,
      floating: true,
      style: {
        backgroundColor: Colors.error,
        alignItems: 'center'
      }
    })
  }

  return (
    <MainContainer>
      {/* TODO - ENLAZAR BUTTON CON RECUPERACION DE CUENTA */}
      <RecoveryButton
        type="clear"
        title={REGISTER.RECOVERY}
        onPress={() => pushScreen(componentId, SCREEN.RECOVERY)}
      />
      <Title>{LOGIN.TITLE}</Title>
      <InputsContainer>
        <InputStyled
          value={pin}
          keyboardType="decimal-pad"
          onChangeText={(value) => setPin(value.replace(/[^0-9]/g, ''))}
          placeholder={LOGIN.PIN}
        />
      </InputsContainer>

      <ContainerBottom>
        <TouchableButton
          title={LOGIN.BUTTON}
          onPress={async () => ((await checkPin(pin)) ? login() : showError())}
        />
      </ContainerBottom>
      <FlashMessage position="top" />
    </MainContainer>
  )
}

Login.options = {
  statusBar: {
    backgroundColor: Colors.mainBackground,
    style: 'dark'
  },
  topBar: {
    visible: false,
    elevation: 0, // Remove border in Android
    borderColor: Colors.mainBackground // Remove border in iOS
  },
  animations: {
    push: {
      content: {
        translationX: {
          from: require('react-native').Dimensions.get('window').width,
          to: 0,
          duration: 300
        }
      }
    },
    pop: {
      content: {
        translationX: {
          from: 0,
          to: require('react-native').Dimensions.get('window').width,
          duration: 300
        }
      }
    }
  }
}

export default Login
