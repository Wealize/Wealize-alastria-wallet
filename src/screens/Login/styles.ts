import { Text, View } from 'react-native'
import styled from 'styled-components'
import { Button, Input } from 'react-native-elements'

import ViewContainer from '../../components/ViewContainer'
import { Colors } from '../../utils/themes'

export const MainContainer = styled(ViewContainer)`
  background-color: #ffffff;
  padding-left: 0;
  padding-right: 0;
`

export const RecoveryButton = styled(Button).attrs({
  containerStyle: {
    marginTop: 25
  },
  buttonStyle: {
    justifyContent: 'flex-start',
    marginLeft: 18
  },
  titleStyle: {
    color: Colors.activeInput,
    fontFamily: 'SourceSansPro',
    fontSize: 14
  },
  icon: {
    name: 'arrow-forward',
    size: 21,
    color: Colors.activeInput
  }
})``

export const Title = styled(Text)`
  font-size: 20px;
  font-weight: bold;
  padding-top: 5%;
  margin-left: 9%;
`

export const InputsContainer = styled(View)`
  margin-left: 6%;
  margin-right: 8%;
  margin-top: 12%;
`

export const InputStyled = styled(Input).attrs({
  selectionColor: Colors.activeInput,
  errorStyle: {
    color: Colors.error
  }
})``

export const ContainerBottom = styled(View)`
  flex: 1;
  justify-content: flex-end;
  align-items: center;
  margin: 20px auto 40px auto;
  width: 100%;
`
