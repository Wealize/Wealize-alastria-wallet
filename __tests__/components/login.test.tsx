import React from 'react'
import renderer from 'react-test-renderer'

import Login from '../../src/screens/Login'

describe('OnboardginComponent', () => {
  it('should render correctly', () => {
    const result = renderer.create(<Login componentId={'1'} />)
    expect(result).toMatchSnapshot()
  })
})
