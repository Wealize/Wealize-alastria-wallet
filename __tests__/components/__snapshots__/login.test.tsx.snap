// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardginComponent should render correctly 1`] = `
<View
  className="sc-bdvvtL sc-gsDKAQ jKiuHH gCheCb"
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        Object {
          "marginTop": 25,
        },
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginLeft": 18,
            "padding": 8,
          }
        }
      >
        <View
          style={
            Object {
              "marginHorizontal": 5,
              "overflow": "hidden",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#00b0ff",
                      "fontSize": 21,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "#00b0ff",
              "fontFamily": "SourceSansPro",
              "fontSize": 14,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Recuperar acceso
        </Text>
      </View>
    </View>
  </View>
  <Text
    className="sc-hKwDye iGOduI"
  >
    Inicio de sesión
  </Text>
  <View
    className="sc-eCImPb hacoEV"
  >
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          className="sc-jRQBWg"
          editable={true}
          keyboardType="decimal-pad"
          onChangeText={[Function]}
          placeholder="Pin de Seguridad"
          placeholderTextColor="#86939e"
          rejectResponderTermination={true}
          selectionColor="#00b0ff"
          style={
            Object {
              "color": "#242424",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <Text
        style={
          Object {
            "color": "#EE3F61",
            "fontSize": 12,
            "margin": 5,
          }
        }
      />
    </View>
  </View>
  <View
    className="sc-gKclnd HYxMd"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ca2b63",
            "borderRadius": 45,
            "height": 39,
            "justifyContent": "center",
            "overflow": "hidden",
            "width": 224,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 16,
            }
          }
        >
          Comenzar
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": Array [
          Object {
            "translateY": -48,
          },
        ],
        "zIndex": 99,
      }
    }
  />
</View>
`;
